File: 1220-count-vowels-permutation/1220-count-vowels-permutation.cpp
class Solution {
public:
    int countVowelPermutation(int n) {
        long aCnt = 1, eCnt = 1, iCnt = 1, oCnt = 1, uCnt = 1, MOD = 1'000'000'007;
        
        for (int i = 1; i < n; ++i){
            long aNewCnt = (eCnt + iCnt + uCnt) % MOD;
            long eNewCnt = (aCnt + iCnt) % MOD;
            long iNewCnt = (eCnt + oCnt) % MOD;
            long oNewCnt = iCnt % MOD;
            long uNewCnt = (iCnt + oCnt) % MOD;
            aCnt = aNewCnt, eCnt = eNewCnt, iCnt = iNewCnt, oCnt = oNewCnt, uCnt = uNewCnt;
        }
        
        return static_cast<int>((aCnt + eCnt + iCnt + oCnt + uCnt) % MOD);
    }
};

File: 1220-count-vowels-permutation/1220-count-vowels-permutation.py
class Solution:
    def countVowelPermutation(self, n: int) -> int:
        permutations = [[1] * n for _ in range(5)]
        MOD = 1_000_000_007
        
        for i in range(1, n):
            # 'a'
            permutations[0][i] = (permutations[1][i - 1] + permutations[2][i - 1] + permutations[4][i - 1]) % MOD
            # 'e'
            permutations[1][i] = (permutations[0][i - 1] + permutations[2][i - 1]) % MOD
            # 'i'
            permutations[2][i] = (permutations[1][i - 1] + permutations[3][i - 1]) % MOD
            # 'o'
            permutations[3][i] = permutations[2][i - 1] % MOD
            # 'u'
            permutations[4][i] = (permutations[2][i - 1] + permutations[3][i - 1]) % MOD
        
        return sum(permutations[i][n - 1] for i in range(5)) % MOD